flow/flow.go

```diff
@@ -37,147 +38,358 @@ func flow(root yaml.Node, env Environment, shouldOverride bool) yaml.Node {
+				}
+				if len(info.RedirectPath) > 0 {
+					redirect = info.RedirectPath
+				}
+				if len(redirect) > 0 {
+					debug.Debug("   REDIRECT -> %v\n", redirect)
+					result=yaml.RedirectNode(result.Value(), result, redirect)
+				}
 
-	switch val := root.Value().(type) {
-	case map[string]yaml.Node:
-		return flowMap(root, env)
-
-	case []yaml.Node:
-		return flowList(root, env)
-
-	case dynaml.Expression:
-		result, ok := val.Evaluate(env)
-		if !ok {
-			return root
+				if replace || info.Replace {
+					debug.Debug("   REPLACE\n")
+					result=yaml.ReplaceNode(result.Value(), result, redirect)
+				} else {
+					if merged || info.Merged {
+						debug.Debug("   MERGED\n")
+						result=yaml.MergedNode(result)
+					}
+				}
+				if expr || result.Merged() || !shouldOverride || result.Preferred() {
+					debug.Debug("   prefer expression over override")
+					debug.Debug("??? ---> %+v\n", result)
+					return result
+				}
+				debug.Debug("???   try override")
+				replace=result.ReplaceFlag()
+				root=result
+			}
+	
+		case string:
+			result := flowString(root, env)
+			if result != nil {
+				_, ok := result.Value().(dynaml.Expression)
+				if ok {
+					// analyse expression before overriding
+					return result
+				}
+			}
 		}
-
-		return result
 	}
 
-	if shouldOverride {
-		overridden, found := env.FindInStubs(env.Path)
+	if !merged && shouldOverride {
+		debug.Debug("/// lookup stub %v -> %v\n",env.Path, env.StubPath)
+		overridden, found := env.FindInStubs(env.StubPath)
 		if found {
-			return overridden
+			root = overridden
+			if keyName != "" {
+				root=yaml.KeyNameNode(root,keyName)
+			}
+    		if replace {
+				return yaml.ReplaceNode(root.Value(),root,redirect)
+			}
+			if redirect != nil {
+				return yaml.RedirectNode(root.Value(),root,redirect)
+			}
+			if merged {
+				return yaml.MergedNode(root)
+			}
 		}
 	}
 
-	_, ok := root.Value().(string)
-	if ok {
-		return flowString(root, env)
-	}
-
 	return root
 }
 
+func optionalMerge(initial bool, node yaml.Node) bool {
+	/////////////////////////////////////////////////////////////
+	// compatibility check. A single merge node is always optional
+	// means: <<: (( merge )) == <<: (( merge || nil ))
+	// the first pass, just parses the dynaml
+	// only the second pass, evaluates a dynaml node!
+	if !initial {
+		merge, ok := node.Value().(dynaml.MergeExpr)
+		return ok && !merge.Required
+	}
+	return false
+	/////////////////////////////////////////////////////////////
+}
+
 func flowMap(root yaml.Node, env Environment) yaml.Node {
+	processed := true
 	rootMap := root.Value().(map[string]yaml.Node)
 
 	env = env.WithScope(rootMap)
-
+	
+	redirect:= root.RedirectPath()
+    replace:= root.ReplaceFlag()
 	newMap := make(map[string]yaml.Node)
 
 	sortedKeys := getSortedKeys(rootMap)
 
+	debug.Debug("HANDLE MAP %v\n", env.Path)
 	// iteration order matters for the "<<" operator, it must be the first key in the map that is handled
```

Comment:
Our apologies, but if you feel like removing this line as well in your next pass through please feel free, otherwise we will be removing it when merging this PR.